src/main/java/com/github/redouane59/twitter/TwitterClient.java

```diff
@@ -399,6 +404,43 @@ public boolean hideReply(final String tweetId, final boolean hide) {
         return result;
     }
 
+    @Override
+    public void startFilteredStream(Consumer<ITweet> consumer){
+        String      url    = this.urlHelper.getFilteredStreamUrl();
+        this.requestHelperV2.executeGetRequestWithConsumer(url, consumer);
+    }
+
+    @Override
+    public List<StreamRule> retrieveFilteredStreamRules() {
+        String       url    = this.urlHelper.getFilteredStreamRulesUrl();
+        StreamRulesDTO result = this.requestHelperV2.executeGetRequest(url, StreamRulesDTO.class).orElseThrow(NoSuchElementException::new);
+        return result.getData();
+    }
+
+    @Override
+    public StreamRule addFilteredStreamRule(String value, String tag){
+        String       url    = this.urlHelper.getFilteredStreamRulesUrl();
+        try {
+            StreamRule rule = StreamRule.builder().value(value).tag(tag).build();
+            String body = "{\"add\": ["+TwitterClient.OBJECT_MAPPER.writeValueAsString(rule)+"]}";
+            StreamRulesDTO result = this.requestHelperV2.executePostRequest(url, body, StreamRulesDTO.class).orElseThrow(NoSuchElementException::new);
+            if(result.getData()==null || result.getData().size()==0) throw new IllegalArgumentException();
+            return result.getData().get(0);
+        }
+        catch (JsonProcessingException e) {
+            LOGGER.error(e.getMessage(), e);
+            throw new IllegalArgumentException();
+        }
+    }
+
+    @Override
+    public StreamMeta deleteFilteredStreamRule(String ruleValue){
```

Comment:
Once again, regarding naming, what is a `rule` but a `String`? Why do you think adding `value` to it makes it more explicit?

```suggestion
    public StreamMeta deleteFilteredStreamRule(String... rules){
```

Reply:
A rule is a little be more complex as the API response shows : 
``` 
    {
      "id": "1165037377523306497",
      "value": "dog has:images",
      "tag": "dog pictures"
    }
```